src/parser/asm_ast.rs

```diff
@@ -5,14 +5,20 @@ use super::ast::{Expression, SelectedExpressions, Statement};
 #[derive(Debug, PartialEq, Eq)]
 pub struct ASMFile(pub Vec<ASMStatement>);
 
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub enum InstructionParams {
+    List(Vec<InstructionParam>),
+    InOut(String, String),
```

Comment:
Right now 1 input 1 output but I can change that if needed

Reply:
Also is it correct that we also need an optional label for each?

Reply:
Yeah, I think the syntax should be something like

`<param> (',' <param>)* ('->' <param> (',' <param>)*)?`

Where

`param = <Identitier> (':' <Identifier>)?`

Then let's fully ditch the old syntax, doesn't really make sense.

This way, you are forced to put the output params last, but I guess that's ok - might go against some asm syntax, though...

If there is at most one output param, you can use functional notation.